core/src/main/scala/org/broadinstitute/dsde/rawls/submissions/SubmissionsService.scala

```diff
@@ -603,8 +603,8 @@ class SubmissionsService(
     val submissionId: UUID = UUID.randomUUID()
 
     for {
-      _ <- requireComputePermission(workspaceName)
       workspaceContext <- getV2WorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write)
+      _ <- accessCheck(workspaceContext, SamWorkspaceActions.compute)
```

Comment:
I moved the `accessCheck` after `getV2WorkspaceContextAndPermissions` to preserve behavior of error conditions. For instance if the current user is not enabled, this will still throw a `UserDisabledException` and if the workspace doesn't exist it will throw a `NoSuchWorkspaceException`. The user-enabled check and workspace-exists checks were the first steps in `requireComputePermission` but are not present in `accessCheck`; they are also the first steps in `getV2WorkspaceContextAndPermissions` so that's why `getV2WorkspaceContextAndPermissions` needs to be first.